test/mock/grpc/mock_management_plane_grpc_server.go

```diff
@@ -50,6 +58,33 @@ func NewManagementGrpcServer() *ManagementGrpcServer {
 		}
 	})
 
+	server.GET("/api/v1/responses", func(c *gin.Context) {
+		if mgs.dataPlaneResponses == nil {
+			c.JSON(http.StatusNotFound, nil)
+		} else {
+			mgs.dataPlaneResponsesMutex.Lock()
+
+			c.JSON(http.StatusOK, gin.H{
+				"dataPlaneResponse": mgs.dataPlaneResponses,
+			})
+
+			mgs.dataPlaneResponsesMutex.Unlock()
+		}
```

Comment:
If we are locking for the read on `mgs.dataPlaneResponses`, then I think the lock is needed also for the first branch of the if-clause:
```suggestion
        mgs.dataPlaneResponsesMutex.Lock()
		if mgs.dataPlaneResponses == nil {
			c.JSON(http.StatusNotFound, nil)
		} else {
			c.JSON(http.StatusOK, gin.H{
				"dataPlaneResponse": mgs.dataPlaneResponses,
			})
		}
		mgs.dataPlaneResponsesMutex.Unlock()
```